src/utils/chat.ts

```diff
@@ -72,3 +93,42 @@ export const buildMessage = <T extends ChatMessage>(
   ...message,
   ts_utc: firebase.firestore.Timestamp.now(),
 });
+
+export interface DivideMessagesReturn<T extends object> {
+  messages: WithId<T>[];
+  allMessagesReplies: WithId<T>[];
+}
+
+export const divideMessages = <T extends ChatMessage>(
+  messages: WithId<T>[]
+): DivideMessagesReturn<T> =>
+  messages.reduce<DivideMessagesReturn<T>>(
+    (acc, message) => {
+      if (message.threadId !== undefined)
+        return {
+          ...acc,
+          allMessagesReplies: [...acc.allMessagesReplies, message],
+        };
+
+      return { ...acc, messages: [...acc.messages, message] };
+    },
+    { messages: [], allMessagesReplies: [] }
+  );
+
+export interface GetMessageRepliesProps<T extends object> {
+  messageId: string;
+  allReplies: WithId<T>[];
+}
+
+export const getMessageReplies = <T extends ChatMessage>({
+  messageId,
+  allReplies,
+}: GetMessageRepliesProps<T>) => {
+  const messageReplies = allReplies.filter(
+    (reply) => reply.threadId === messageId
+  );
+
+  if (messageReplies.length === 0) return undefined;
+
+  return messageReplies;
+};
```

Comment:
Is there any reason we need to have this as `undefined` when there are none? In most cases an empty array is more useful than `undefined`.

Reply:
Fixed in https://github.com/sparkletown/sparkle/pull/1374/commits/77d4d4d3a8ed73f8dcab26a794b800f92c74d2ec